xds/src/main/java/io/grpc/xds/client/ControlPlaneClient.java

```diff
@@ -152,9 +152,8 @@ void adjustResourceSubscription(XdsResourceType<?> resourceType) {
       startRpcStream();
     }
     Collection<String> resources = resourceStore.getSubscribedResources(serverInfo, resourceType);
-    if (resources != null) {
-      adsStream.sendDiscoveryRequest(resourceType, resources);
-    }
+    adsStream.sendDiscoveryRequest(resourceType,
```

Comment:
If we cleanup the nonce, then there is no need to send this empty discovery request to get the new nonce.

Reply:
Considering the eventual consistent, I think it is correct. According to the Istio server,

https://github.com/istio/istio/blob/master/pkg/xds/server.go#L359-L363

If an empty subscription has been received (before nonce check), the resource will be recycled. I think this is the only point.

Reply:
As I mentioned at https://github.com/grpc/grpc-java/pull/11264#issuecomment-2249385118 .

We aren't sending a request to get a new nonce here. In the case there are no resources. This is updating the subscription to be "no resources" just before the resource type is deleted. We do need it.